qcfractal/interface/models/rest_models.py

```diff
@@ -0,0 +1,221 @@
+    id = "id"
+    molecule_hash = "molecule_hash"
+    molecular_formula = "molecular_formula"
+
+
+class MoleculeGETBody(BaseModel):
+    class Meta(BaseModel):
+        index: MoleculeIndices
+
+    data: List[str]
+    meta: Meta
+
+
+class MoleculeGETResponse(BaseModel):
+    meta: ResponseGETMeta
+    data: List[Molecule]
+
+    class Config:
+        json_encoders = {**ndarray_encoder}
+
+
+class MoleculePOSTBody(BaseModel):
+    meta: Dict = None
+    data: Dict[str, Molecule]
+
+    class Config:
+        json_encoders = {**ndarray_encoder}
+
+
+class MoleculePOSTResponse(BaseModel):
+    meta: ResponsePOSTMeta
+    data: Dict[str, str]
+
+
+### Options
+
+class OptionGETBody(BaseModel):
+    meta: Dict = None
+    data: Dict[str, Any]
+
+
+class OptionGETResponse(BaseModel):
+    meta: ResponseGETMeta
+    data: List[Dict[str, Any]]
+
+
+class OptionPOSTBody(BaseModel):
+    meta: Dict = None
+    data: List[Dict[str, Any]]
+
+    @validator("data", whole=True, pre=True)
+    def cast_dict_to_list_of_dict(cls, v):
+        if isinstance(v, dict):
+            return [v]
+        return v
+
+
+class OptionPOSTResponse(BaseModel):
+    data: List[str]
+    meta: ResponsePOSTMeta
+
+
+### Collections
+
+class CollectionGETBody(BaseModel):
+    class Data(BaseModel):
+        collection: str = None
+        name: str = None
+
+        @validator("collection")
+        def cast_to_lower(cls, v):
+            return v.lower()
+
+    meta: Dict = None
+    data: Data
+
+
+class CollectionGETResponse(BaseModel):
+    meta: ResponseGETMeta
+    data: List[Dict[str, Any]]
+
+    @validator("data", whole=True)
+    def ensure_collection_name_in_data_get_res(cls, v):
+        for col in v:
+            if "name" not in col or "collection" not in col:
+                raise ValueError("Dicts in 'data' must have both 'collection' and 'name'")
+        return v
+
+
+class CollectionPOSTBody(BaseModel):
+    class Meta(BaseModel):
+        overwrite: bool = False
+
+    class Data(BaseModel):
+        id: str = "local"  # Auto blocks overwriting in mongoengine_socket
+        collection: str
+        name: str
+
+        @validator("collection")
+        def cast_to_lower(cls, v):
+            return v.lower()
+
+        class Config:
+            # Maps effectively Dict[str, Any] but enforces the collection and name fields
+            allow_extra = True
+
+    meta: Meta = Meta()
+    data: Data
+
+
+class CollectionPOSTResponse(BaseModel):
+    data: Union[str, None]
+    meta: ResponsePOSTMeta
+    
+    
+### Result
+
+
+class ResultGETBody(BaseModel):
+    class Meta(BaseModel):
+        projection: Dict[str, Any] = None
+
+    meta: Meta = Meta()
+    data: Dict[str, Any]
+
+    @validator("data", whole=True)
+    def only_data_keys(cls, v):
+        keys = ["program", "molecule", "driver", "method", "basis", "options", "hash_index", "task_id", "id", "status"]
+        final_data = {key: v[key] for key in keys if key in v}
```

Comment:
Can we do something like:

```
valid_keys = {"program", ...}
final_data = {key: v[key] for key in (v.keys() & valid_keys)}
```

I feel like we miss out on set manipulation too much!